src/components/SearchWidgetsWrapper.tsx

```diff
@@ -0,0 +1,98 @@
+import React, { useState } from 'react'
+import { SearchWidget } from './SearchWidget'
+import { SalesEvidenceSearchWidget } from './salesEvidenceSearch'
+import { DEFAULT_SEARCH_PARAMS, IntersectionFilterParams } from '../types'
+import { Test3 } from './test3'
+
+type SearchWidgetProps = {
+    searchParameters: IntersectionFilterParams[]
+    changeSearchParameters: (newParameters: IntersectionFilterParams[]) => void
+    setReadyForSearch: (isReady: boolean) => void
+}
+
+export function SearchWidgetWrapper (props: SearchWidgetProps) {
+    const [widgetStateChange, setWidgetStateChange] = useState<boolean>(false)
+    const [widgetStateChangeSales, setWidgetStateChangeSales] = useState<boolean>(false)
+    const [widgetStateChangeLease, setWidgetStateChangeLease] = useState<boolean>(false)
+
+    function hideWidget (componentName: any) {
+        switch (componentName) {
+        case 'widgetStateChange':
+            setWidgetStateChange(!widgetStateChange)
+            break
+        case 'widgetStateChangeSales':
+            setWidgetStateChangeSales(!widgetStateChangeSales)
+            break
+        case 'widgetStateChangeLease':
+            setWidgetStateChangeLease(!widgetStateChangeLease)
+            break
+        default:
+            setWidgetStateChange(!widgetStateChange)
+        }
+    }
+
+    return (
+        <div>
+            {widgetStateChange && props.searchParameters.map((searchParameters, idx) => {
+                return (
+                    <div className="search-params-wrapper" key={searchParameters.id}>
+                        <SearchWidget
+                            searchParameters={searchParameters}
+                            changeSearchParameters={(newSearchParams) => {
+                                const updatedSearchParams = [...props.searchParameters]
+                                updatedSearchParams[idx] = newSearchParams
+                                props.changeSearchParameters(updatedSearchParams)
+                            }}
+                        />
+                        <button className="danger" onClick={() => {
+                            const updatedSearchParams = [...props.searchParameters]
+                            updatedSearchParams.splice(idx, 1)
+                            props.changeSearchParameters(updatedSearchParams)
+                        }}>Remove Search Group
+                        </button>
+                    </div>
+                )
+            })}
+            {widgetStateChangeSales && props.searchParameters.map((searchParameters, idx) => {
```

Comment:
For the sales and lease evidence filters we don't need to have multiple search properties so you can get rid of `.map()` and just directly access the param object in the array `props.searchParameters[0]`.

Reply:
I can't seem to get it to work without the mapping (including the button to remove search group)